sdk/core/core-http/.eslintrc.json

```diff
@@ -1,7 +1,12 @@
 {
   "plugins": ["@azure/azure-sdk"],
-  "extends": ["../../.eslintrc.old.json", "plugin:@azure/azure-sdk/recommended"],
-  "parserOptions": {
-    "createDefaultProgram": true
+  "extends": ["plugin:@azure/azure-sdk/azure-sdk-base"],
+  "rules": {
+    "@typescript-eslint/no-unused-vars": "off",
```

Comment:
how bad it was to turn on "no-unused-vars"?

Reply:
These are all of the eslint issues related to `no-unused-vars`

```
/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/src/browserFetchHttpClient.ts
   9:18  warning  '_httpRequest' is defined but never used        @typescript-eslint/no-unused-vars
  13:18  warning  '_operationResponse' is defined but never used  @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/src/policies/disableResponseDecompressionPolicy.browser.ts
  23:14  warning  '_nextPolicy' is defined but never used  @typescript-eslint/no-unused-vars
  23:42  warning  '_options' is defined but never used     @typescript-eslint/no-unused-vars
  35:28  warning  '_request' is defined but never used     @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/src/policies/proxyPolicy.browser.ts
  16:41  warning  '_proxyUrl' is defined but never used       @typescript-eslint/no-unused-vars
  20:29  warning  '_proxySettings' is defined but never used  @typescript-eslint/no-unused-vars
  22:14  warning  '_nextPolicy' is defined but never used     @typescript-eslint/no-unused-vars
  22:42  warning  '_options' is defined but never used        @typescript-eslint/no-unused-vars
  34:22  warning  '_request' is defined but never used        @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/src/policies/throttlingRetryPolicy.ts
  70:39  warning  '_' is defined but never used  @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/test/defaultHttpClientTests.browser.ts
  35:46  warning  '_url' is defined but never used     @typescript-eslint/no-unused-vars
  35:52  warning  '_method' is defined but never used  @typescript-eslint/no-unused-vars
  35:61  warning  '_body' is defined but never used    @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/test/defaultHttpClientTests.node.ts
  104:46  warning  '_url' is defined but never used     @typescript-eslint/no-unused-vars
  104:52  warning  '_method' is defined but never used  @typescript-eslint/no-unused-vars
  104:61  warning  '_body' is defined but never used    @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/test/defaultHttpClientTests.ts
  162:43  warning  '_url' is defined but never used      @typescript-eslint/no-unused-vars
  162:49  warning  '_method' is defined but never used   @typescript-eslint/no-unused-vars
  162:58  warning  '_body' is defined but never used     @typescript-eslint/no-unused-vars
  238:58  warning  '_headers' is defined but never used  @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/test/mockHttp.ts
  50:15  warning  '_url' is defined but never used  @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/test/policies/bearerTokenAuthenticationPolicyTests.ts
  105:5  warning  '_scopes' is defined but never used   @typescript-eslint/no-unused-vars
  106:5  warning  '_options' is defined but never used  @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/test/policies/disableResponseDecompressionPolicy.browser.ts
  16:19  warning  '_' is defined but never used  @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/test/policies/proxyPolicyTests.browser.ts
  21:19  warning  '_' is defined but never used  @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/test/policies/proxyPolicyTests.node.ts
  22:19  warning  '_' is defined but never used  @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/test/policies/systemErrorRetryPolicyTests.spec.ts
  144:30  warning  '_request' is defined but never used  @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/test/policies/throttlingRetryPolicyTests.ts
  83:72  warning  '_' is defined but never used  @typescript-eslint/no-unused-vars

/home/vsonline/workspace/azure-sdk-for-js/sdk/core/core-http/test/policies/tracingPolicyTests.ts
  43:13  warning  '_key' is defined but never used      @typescript-eslint/no-unused-vars
  43:27  warning  '_value' is defined but never used    @typescript-eslint/no-unused-vars
  46:15  warning  '_key' is defined but never used      @typescript-eslint/no-unused-vars
  49:13  warning  '_key' is defined but never used      @typescript-eslint/no-unused-vars
  81:13  warning  '_name' is defined but never used     @typescript-eslint/no-unused-vars
  81:28  warning  '_options' is defined but never used  @typescript-eslint/no-unused-vars
```

Reply:
Variables that start with `_` _should_ be ignored. Maybe we need to configure the linting rule to use that ignore pattern:

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md

and

https://eslint.org/docs/rules/no-unused-vars#options

Reply:
`{ "argsIgnorePattern": "^_" }` configure it with this, preferably globally

Reply:
I tried using:

```ts
"@typescript-eslint/no-unused-vars": ["warn", { "varsIgnorePattern": "^_" }],
```

In the file `common/tools/eslint-plugin-azure-sdk/src/configs/azure-sdk-base.ts`, and removing `"@typescript-eslint/no-unused-vars": "off",` from the .eslintrc.json file in the core-http folder, but it doesn't seem to be working.

What am I missing? 🤔 
Am

Reply:
hm, I would think that would work? Did you recompile `eslint-plugin-azure-sdk` after changing it?

Reply:
ok, for arguments, we need to configure `"argsIgnorePattern"`, not `varsIngorePattern`

Reply:
I pushed a fix for this.